tensorflow/core/kernels/mkl/mkl_conv_ops.cc

```diff
@@ -944,23 +944,18 @@ class MklConvOp : public OpKernel {
       if (native_format) {
         // Forward the summand tensor to the output only if it has no other
         // references, otherwise make a copy of it.
-        if (!context->forward_input_to_output_with_shape(
+        if (context->forward_input_to_output_with_shape(
                 kInputIndex_Add, kOutputIndex_Dst, output_tf_shape,
                 output_tensor)) {
-          AllocateOutputSetMklShape(context, kOutputIndex_Dst, output_tensor,
-                                    output_tf_shape, *output_mkl_shape,
-                                    native_format);
-          bool result =
-              (*output_tensor)->CopyFrom(add_tensor, add_tensor.shape());
-          DCHECK(result);
+          return;
         }
-        return;
       }
       // Check if reorder is needed
       if (add_mkl_shape == *output_mkl_shape &&
           ForwardMklTensorInToOutWithMklShape(context, kInputIndex_Add,
                                               kOutputIndex_Dst, output_tensor,
-                                              add_mkl_shape, false)) {
+                                              add_mkl_shape, false) &&
+          !native_format) {
```

Comment:
Nit: Can `!native_format` be first? It's the cheapest check.
```c++
      if (!native_format && 
          add_mkl_shape == *output_mkl_shape &&
          ForwardMklTensorInToOutWithMklShape(context, kInputIndex_Add,
                                              kOutputIndex_Dst, output_tensor,
                                              add_mkl_shape, false)) {
```

Reply:
Done.